ArcGISRuntimeSDKQt_CppSamples/Scenes/CreateTerrainSurfaceFromLocalRaster/CreateTerrainSurfaceFromLocalRaster.cpp

```diff
@@ -0,0 +1,104 @@
+// [WriteFile Name=CreateTerrainSurfaceFromLocalRaster, Category=Scenes]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "CreateTerrainSurfaceFromLocalRaster.h"
+
+#include "Scene.h"
+#include "SceneQuickView.h"
+#include "RasterElevationSource.h"
+
+#include <QUrl>
+#include <QDir>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace {
+  QString defaultDataPath()
+  {
+    QString dataPath;
+
+    #ifdef Q_OS_ANDROID
+      dataPath = "/sdcard";
+    #elif defined Q_OS_IOS
+      dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+    #else
+      dataPath = QDir::homePath();
+    #endif
+
+    return dataPath;
+  }
+}
+
+CreateTerrainSurfaceFromLocalRaster::CreateTerrainSurfaceFromLocalRaster(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(Basemap::imagery(this), this))
+{
+  // create the MontereyElevation data path
+  // data is downloaded automatically by the sample viewer app. Instructions to download
+  // separately are specified in the readme.
+  const auto montereyRasterElevationPath = QStringList{defaultDataPath() + "/ArcGIS/Runtime/Data/raster/MontereyElevation.dt2"};
+
+  //Create the elevation source from the local raster(s). RasterElevationSource can take multiple files as inputs, but in this case only takes one.
+  auto* elevationSrc = new RasterElevationSource{montereyRasterElevationPath, this};
```

Comment:
normally we don't use `auto` in sample code

Reply:
I also wonder if we should give this guy some more prominence in the sample somehow - since this is the part of the API we are demonstrating? For example, we could log a message when it is loaded or something...